### ISSUE TITLE
SDK resolves paths differently internally. high bug risk. 

### ISSUE THREAD

**Issue by samuelstroschein** [opral/monorepo#2011]
## Problem 

The SDK uses different multiple filesystem instances that resolve paths differently, leading to regression bugs. 

_`fs` will resolve from `cwd` or the root of a repo._

https://github.com/opral/monorepo/blob/9eabe589795df8cd8b869685dfb76133bb58e7b2/inlang/source-code/sdk/src/loadProject.ts#L70

_this instance will resolve from the `projectPath`_
https://github.com/opral/monorepo/blob/9eabe589795df8cd8b869685dfb76133bb58e7b2/inlang/source-code/sdk/src/loadProject.ts#L98-L101

## Proposal

Force resolution from the `projectPath` as decided in https://github.com/opral/monorepo/issues/1441 and implemented in https://github.com/opral/monorepo/pull/1446 to avoid bugs.

**Comment by samuelstroschein** [comment]
@jldec assigning you to make sure this will be resolved. 

PS this might be a good first issue

**Comment by janfjohannes** [comment]
also this might make sense to do in lix @samuelstroschein  

- first enforce use of repo / llix everywhere
- next fs is always made with a virtual root at the repo 

makes more sense to me than having this in the sdk at all, what do you think?

**Comment by samuelstroschein** [comment]
@janfjohannes yes and no. having `repo.fs` to resolve from the repo root makes sense. But, inlang modules need to resolve relatively from the project rather than the repo root. 

_The example paths beneath are resolved from the project directory. Which makes sense. The project (settings) have no knowledge of a repo root_. 

https://github.com/opral/monorepo/blob/ed859116c0eae507a44f166e72b956956b0afba3/inlang/source-code/end-to-end-tests/inlang-nextjs/project.inlang/settings.json#L5-L18

https://github.com/opral/monorepo/blob/ed859116c0eae507a44f166e72b956956b0afba3/project.inlang/settings.json#L23

https://github.com/opral/monorepo/blob/ed859116c0eae507a44f166e72b956956b0afba3/inlang/source-code/paraglide/paraglide-js-adapter-next/example-app/project.inlang/settings.json#L16

**Comment by felixhaeberle** [comment]
This is an ongoing issue with the VSCode extension too. 

Imagine there is a repository on my machine, but I opened a subfolder in VSCode — there is no way I can find out the correct relative path from this opened subfolder to non existent/not specified repo root to my project, e.g. ./project.inlang (when opened in ./frontend).

The correct path from a repository level would be ./frontend/project.inlang — but when opened in a subfolder, this will not work anymore with linking with the correct project path to Fink, as Fink can't resolve only ./project.inlang (in ./frontend) because Fink checks out the repository at the root level and then needs the full project path e.g. ./frontend/project.inlang

Interesting: When you do not open your repository at the root, the built-in Git functionality in VSCode is then also not working. So this "bug" is expected currently — but we can probably do better.

**Comment by jldec** [comment]
closing this in favor of LIX-54
